src/Authenticators/TOTPAuthenticator.php

```diff
@@ -26,47 +28,54 @@ class TOTPAuthenticator extends BootstrapMFAAuthenticator
      * a specific TOTP authenticator
      */
     private static $algorithm = 'sha1';
-
-
+    
     /**
-     * @param string $link
-     * @return \SilverStripe\Security\MemberAuthenticator\LoginHandler|static
+     * Get configured algorithm for TOTP Authenticator
+     *
+     * Must be one of: "sha1", "sha256", "sha512"
+     * If not specified or invalid, default to "sha1"
+     * @return string
      */
-    public function getLoginHandler($link)
+    public static function get_algorithm()
     {
-        return TOTPLoginHandler::create($link, $this);
+        $algorithm = self::config()->get('algorithm');
```

Comment:
This should probably be `static::config()` to allow for late static binding & the ability for subclasses to have their own specific config.